test/proxy-upgrades/enable-guards.ts

```diff
@@ -0,0 +1,394 @@
+      nonce: await userSCW.getNonce(1),
+    });
+
+    const chainId = await userSCW.getChainId();
+    const { signer, data } = await safeSignTypedData(
+      accounts[0],
+      userSCW,
+      safeTx,
+      chainId
+    );
+
+    const transaction: Transaction = {
+      to: safeTx.to,
+      value: safeTx.value,
+      data: safeTx.data,
+      operation: safeTx.operation,
+      targetTxGas: safeTx.targetTxGas,
+    };
+    const refundInfo: FeeRefund = {
+      baseGas: safeTx.baseGas,
+      gasPrice: safeTx.gasPrice,
+      tokenGasPriceFactor: safeTx.tokenGasPriceFactor,
+      gasToken: safeTx.gasToken,
+      refundReceiver: safeTx.refundReceiver,
+    };
+
+    let signature = "0x";
+    signature += data.slice(2);
+    await expect(
+      userSCW
+        .connect(accounts[0])
+        .execTransaction_S6W(transaction, refundInfo, signature)
+    ).to.emit(userSCW, "ExecutionSuccess");
+
+    expect(await token.balanceOf(charlie)).to.equal(
+      balanceCharlieBefore.add(amountToTransfer)
+    );
+
+  });
+
+  it("Should revert for delegatecall to unapproved contract", async function () {
+
+    const IncrNonceLib = await ethers.getContractFactory("TestIncreaseNonceLib");
+    const trustedTarget = await IncrNonceLib.deploy();
+    await trustedTarget.deployed();
+
+    const nonTrustedTarget = await IncrNonceLib.deploy();
+    await nonTrustedTarget.deployed();
+
+    const DelegateCallGuard = await ethers.getContractFactory(
+      "DelegateCallTransactionGuard"
+    );
+    const dcGuard = await DelegateCallGuard.deploy(trustedTarget.address);
+    await dcGuard.deployed();
+
+    await executeContractCallWithSigners(userSCW, userSCW, "setGuard",[dcGuard.address], [accounts[0]]);
+
+    const delegateCall = 1;
+
+    const safeTx: SafeTransaction = buildSafeTransaction({
+      to: nonTrustedTarget.address,
+      operation: delegateCall,
+      data: nonTrustedTarget.interface.encodeFunctionData("increaseNonce", [2]),
+      nonce: await userSCW.getNonce(1),
+    });
+
+    const chainId = await userSCW.getChainId();
+    const { signer, data } = await safeSignTypedData(
+      accounts[0],
+      userSCW,
+      safeTx,
+      chainId
+    );
+
+    const transaction: Transaction = {
+      to: safeTx.to,
+      value: safeTx.value,
+      data: safeTx.data,
+      operation: safeTx.operation,
+      targetTxGas: safeTx.targetTxGas,
+    };
+    const refundInfo: FeeRefund = {
+      baseGas: safeTx.baseGas,
+      gasPrice: safeTx.gasPrice,
+      tokenGasPriceFactor: safeTx.tokenGasPriceFactor,
+      gasToken: safeTx.gasToken,
+      refundReceiver: safeTx.refundReceiver,
+    };
+
+    let signature = "0x";
+    signature += data.slice(2);
+
+    await expect(
+      userSCW
+        .connect(accounts[0])
+        .execTransaction_S6W(transaction, refundInfo, signature)
+    ).to.be.revertedWith("DelegateCallGuardRestricted");
+
+  });
+
+  it("Should allow delegatecall to approved contract", async function () {
```

Comment:
add case for should not allow dc to unapproved contract

Reply:
there is one, it's called "Should revert for delegatecall to unapproved contract"